tools/parse_baselines/main.py

```diff
@@ -132,41 +132,33 @@ def main():
     )
     args = parser.parse_args()
 
-    # Create the concatenated data file.
-    data_file = concatenate_data_files(get_data_files(args))
-
-    # Instantiate a file data provider.
-    data_provider = FileDataProvider(data_file.name)
-
     # Instantiate the right data parser.
-    parser = DATA_PARSERS[args.test](data_provider)
+    parser = DATA_PARSERS[args.test](read_data_files(args))
 
     # Finally, parse and update the baselines.
-    with open(
-        f"./tests/integration_tests/performance/configs/"
-        f"test_{args.test}_config_{args.kernel}.json",
-        "r+",
-        encoding="utf8",
-    ) as baselines_file:
-        json_baselines = json.load(baselines_file)
-        current_cpus = json_baselines["hosts"]["instances"][args.instance]["cpus"]
-        cpus = parser.parse()
-
-        for cpu in cpus:
-            model = cpu["model"]
-            for old_cpu in current_cpus:
-                if old_cpu["model"] == model:
-                    old_cpu["baselines"] = cpu["baselines"]
-        baselines_file.truncate(0)
-        baselines_file.seek(0, 0)
-        json.dump(json_baselines, baselines_file, indent=4)
-
-        # Warn against the fact that not all CPUs pertaining to
-        # some arch were updated.
-        assert len(cpus) == len(current_cpus), (
-            "It may be that only a subset of CPU types were updated! "
-            "Need to run again! Nevertheless we updated the baselines..."
-        )
+    baselines_path = Path(
+        f"./tests/integration_tests/performance/configs/test_{args.test}_config_{args.kernel}.json"
+    )
+    json_baselines = json.loads(baselines_path.read_text("utf-8"))
+    current_cpus = json_baselines["hosts"]["instances"][args.instance]["cpus"]
+    cpus = parser.parse()
+
+    for cpu in cpus:
+        model = cpu["model"]
+        for old_cpu in current_cpus:
```

Comment:
nit: The variable name (`old_cpu` comes from `current_cpus`) looks strange for me. How about changing `current_cpus` to `old_cpus` on this occasion?

Reply:
Sounds good... I can do it in a follow-up or tomorrow if it doesn't get merged before then.